ResearchSuite/ResearchSuite/RSDFormStepDataSourceObject.swift

```diff
@@ -0,0 +1,843 @@
+     */
+    public var isAnswerValid: Bool {
+        return true
+    }
+    
+    public init(items: [RSDTableItem], beginningRowIndex: Int) {
+        self.items = items
+        self.beginningRowIndex = beginningRowIndex
+    }
+}
+
+/**
+ `RSDTableItemGroup` is used to represent a single input field.
+ */
+open class RSDInputFieldTableItemGroup : RSDTableItemGroup {
+    
+    public let inputField: RSDInputField
+    public let uiHint: RSDFormUIHint
+    public let answerType: RSDAnswerResultType
+    public let defaultAnswer: Any
+    public let textFieldOptions: RSDTextFieldOptions?
+    
+    open fileprivate(set) var numberRange: RSDDecimalRange?
+    open private(set) var formatter: Formatter?
+    open private(set) var pickerSource: RSDPickerDataSource?
+    
+    public init(beginningRowIndex: Int, inputField: RSDInputField, uiHint: RSDFormUIHint, answerType: RSDAnswerResultType, defaultAnswer: Any? = nil, textFieldOptions: RSDTextFieldOptions? = nil, items: [RSDTableItem]? = nil, formatter: Formatter? = nil, pickerSource: RSDPickerDataSource? = nil) {
+        self.inputField = inputField
+        self.uiHint = uiHint
+        self.answerType = answerType
+        self.pickerSource = pickerSource
+        self.formatter = formatter
+        self.defaultAnswer = defaultAnswer ?? NSNull()
+        
+        // Set the text field options
+        self.textFieldOptions = textFieldOptions ?? inputField.textFieldOptions ?? {
+            switch answerType.baseType {
+            case .decimal:
+                return RSDTextFieldOptionsObject(keyboardType: .decimalPad)
+            case .integer, .timeInterval:
+                return RSDTextFieldOptionsObject(keyboardType: .numberPad)
+            case .date, .string:
+                return RSDTextFieldOptionsObject(keyboardType: .default)
+            case .boolean, .data:
+                return nil
+            }
+        }()
+        
+        super.init(items: items ?? [RSDInputFieldTableItem(rowIndex: beginningRowIndex, inputField: inputField)], beginningRowIndex: beginningRowIndex)
+    }
+
+    /**
+     Convenience property for accessing the identifier associated with the item group.
+     */
+    public var identifier: String {
+        return inputField.identifier
+    }
+    
+    /**
+     Save an answer for this ItemGroup. This is used only for those questions that have single answers,
+     such as text and numeric answers, as opposed to booleans or text choice answers.
+     */
+    public var answer: Any {
+        return _answer ?? defaultAnswer
+    }
+    fileprivate var _answer: Any?
+    
+    /**
+     The text string to display as the answer.
+     */
+    public var answerText: String? {
+        return (_answer as? String) ?? formatter?.string(for: _answer)
+    }
+    
+    /**
+     Set the new answer value. This will throw an error if the value isn't valid. Otherwise, it will set the answer.
+     @param newValue    The new value for the answer.
```

Comment:
We should probably try to be consistent with whether or not @param and @return descriptions end with a period or not, at least within a file.

Reply:
I am trying to be, but some of this is copy/paste from Josh Bruin and I didn't catch all the typos. I've been try to always use the period at the end.